server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanDynamicPropertyCache.java

```diff
@@ -0,0 +1,118 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.infinispan;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.Resource;
+
+import org.hawkular.apm.api.services.ServiceLifecycle;
+import org.hawkular.apm.api.utils.PropertyUtil;
+import org.hawkular.apm.server.api.services.CacheException;
+import org.hawkular.apm.server.api.services.DynamicPropertyService;
+import org.hawkular.apm.server.api.services.ObservableProperty;
+import org.infinispan.Cache;
+import org.infinispan.manager.CacheContainer;
+
+/**
+ * @author Pavol Loffay
+ */
+public class InfinispanDynamicPropertyCache implements DynamicPropertyService, ServiceLifecycle {
+
+    private static final Logger log = Logger.getLogger(InfinispanSpanCache.class.getName());
+
+
+    protected static final String CACHE_NAME = "property";
+
+    @Resource(lookup = "java:jboss/infinispan/APM")
+    private CacheContainer cacheContainer;
+
+    private Cache<String, ObservableProperty> propertyCache;
+
+
+    @Override
+    @PostConstruct
+    public void init() {
+        // If cache container not already provisions, then must be running outside of a JEE
+        // environment, so create a default cache container
+        if (cacheContainer == null) {
+            log.fine("Using default cache");
+            setPropertyCache(InfinispanCacheManager.getDefaultCache(CACHE_NAME));
+        } else {
+            log.fine("Using container provided cache");
+            setPropertyCache(cacheContainer.getCache(CACHE_NAME));
+        }
+    }
+
+    @Override
+    public ObservableProperty<?> get(String tenantId, String id) {
+        ObservableProperty<?> observableProperty = propertyCache.get(id);
+
+        if (observableProperty == null) {
+            observableProperty = getFromSystemOrEnvVariables(id);
+            if (observableProperty == null) {
+                return null;
+            }
+
+            try {
+                store(tenantId, Collections.singletonList(observableProperty));
+            } catch (CacheException ex) {
+                log.severe(String.format("Could not store system or env property: %s, to cache: ", id));
+            }
+        }
+
+        log.finest("Get communication details [id=" + id + "] = " + observableProperty);
+        return observableProperty;
+    }
+
+    @Override
+    public void store(String tenantId, List<ObservableProperty> observableProperties) throws CacheException {
+        if (cacheContainer != null) {
+            propertyCache.startBatch();
```

Comment:
It might make sense. This store method is from the `Cache` interface. 
https://github.com/hawkular/hawkular-apm/blob/master/server/api/src/main/java/org/hawkular/apm/server/api/services/Cache.java